src/sys/socket/addr.rs

```diff
@@ -64,6 +64,24 @@ impl InetAddr {
         }
     }
 
+    /// Creates an `IpAddr` struct from libc's sockaddr.
+    ///
+    /// Note that this supports only IPv4 and IPv6 address families.
+    pub unsafe fn from_libc_sockaddr(addr: *mut libc::sockaddr) -> Option<InetAddr> {
```

Comment:
Why is this unsafe and the other `from_libc` isn't? What's the reason for providing this API?

Reply:
If it's not a public function, you should remove the `pub`